conans/client/tools/oss.py

```diff
@@ -313,6 +313,9 @@ def get_gnu_triplet(os, arch, compiler=None):
     :param compiler: compiler used to create the triplet (only needed fo windows)
     """
 
+    if os == "Windows" and compiler is None:
+        raise ConanException("Compiler not specified and needed for os=\"Windows\"")
```

Comment:
better:  'compiler' parameter for 'get_gnu_triplet' function is not specified and needed for os=Windows